### PR TITLE
doc: security best practices and testing docs

### PR DESCRIPTION
**Pull request by gyrationtechs**
New docs:
- Smart contract security
- Testing and debugging

Hide the Security section in the Concepts category to avoid creating duplicate docs

### CODE REVIEW

**Review by nstanford5 — CHANGES_REQUESTED** [review]

File: docs/compact/testing.mdx [hunk 1]
```diff
@@ -1,10 +1,460 @@
...
\ No newline at end of file
+Testing Compact smart contracts requires a comprehensive approach that validates both functional correctness and security properties. 
+
+This guide covers unit testing, integration testing, and debugging strategies for privacy-preserving contracts on Midnight.
+
+## Overview
+
+Compact contracts execute across multiple contexts (on-chain ledger, zero-knowledge circuits, and local witnesses), requiring different testing strategies for each layer. Effective testing ensures your contract behaves correctly, maintains privacy guarantees, and handles edge cases appropriately.
+
+### Test layers
```
Comment (DianaHackmamba) [inline docs/compact/testing.mdx]: Do not have just one subheading.

**Review by DianaHackmamba — DISMISSED** [review]

**Review by nstanford5 — DISMISSED** [review]

**Review by nstanford5 — APPROVED** [review]
